### PR TITLE
Profilepage

### PR DESCRIPTION
**Pull request by marijadmitrovic**
Diego , mira esto cuando puedas :)

### CODE REVIEW

File: src/Form/CreateUserType.php [hunk 1]
```diff
@@ -22,11 +22,11 @@ public function buildForm(FormBuilderInterface $builder, array $options)
             ->add('name',TextType::class)
             ->add('lastname',TextType::class)
             ->add('email',EmailType::class)
-            ->add('plainPassword', RepeatedType::class, array('type' => PasswordType::class,
-                'first_options'  => array('label' => 'Password'),
-                'second_options' => array('label' => 'Repeat Password'),
+            ->add('plainPassword', RepeatedType::class, ['type' => PasswordType::class,
+                'first_options'  => ['label' => 'Password'],
+                'second_options' => ['label' => 'Repeat Password'],
                 'attr' => array('id'=>'password1')
```
Comment (AnkDev) [inline src/Form/CreateUserType.php]: change the array() like the other with []

File: src/Twig/TwigExtension.php [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        return [
+            new TwigFunction('is_current_path', [$this, 'getIsCurrent'], ['is_safe_callback' => [$this, 'isUrlGenerationSafe']]),
+        ];
+    }
+
+    /**
+     * @param string $name
+     * @param array  $parameters
+     * @param bool   $relative
+     *
```
Comment (AnkDev) [inline src/Twig/TwigExtension.php]: you can delete this line

File: src/Twig/TwigExtension.php [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+     * @return string
+     */
+    public function getIsCurrent($name, $parameters = array(), $relative = false)
+    {
+        $path = $this->router->generate($name, $parameters, $relative ? UrlGeneratorInterface::RELATIVE_PATH : UrlGeneratorInterface::ABSOLUTE_PATH);
+        $currentPath = $this->request->getRequestUri();
+
```
Comment (AnkDev) [inline src/Twig/TwigExtension.php]: delete this line :D

File: templates/base.html.twig [hunk 4]
```diff
@@ -1,94 +1,65 @@
...
-                            {% endif %}
-                        </ul>
-                    </div>
-                </nav>
-            </div>
+    </div>
+    <!--Mainmenu-area-->
+    <div class="mainmenu-area" data-spy="affix" data-offset-top="100">
+        <div class="container">
+
```
Comment (AnkDev) [inline templates/base.html.twig]: Delete this line

File: templates/base.html.twig [hunk 5]
```diff
@@ -1,94 +1,65 @@
...
-        <script src="{{ asset('js/jquery.parallax-1.1.3.js') }}"></script>
-        <script src="{{ asset('js/scrollUp.min.js') }}"></script>
-        <script src="{{ asset('js/magnific-popup.min.js') }}"></script>
-        <script src="{{ asset('js/wow.min.js') }}"></script>
-        <!--Main-active-JS-->
-        <script src="{{ asset('js/main.js') }}"></script>
-    {% endblock %}
-    </body>
+{% block body %} {% endblock %}
+
```
Comment (AnkDev) [inline templates/base.html.twig]: Delete

File: templates/user_profile/edit.html.twig [hunk 6]
```diff
@@ -0,0 +1,57 @@
+{% extends 'user_profile/layout.html.twig' %}
+
```
Comment (AnkDev) [inline templates/user_profile/edit.html.twig]: delete :)

File: templates/user_profile/index.html.twig [hunk 7]
```diff
@@ -0,0 +1,35 @@
+{% extends 'user_profile/layout.html.twig' %}
+
```
Comment (AnkDev) [inline templates/user_profile/index.html.twig]: PLS

File: templates/user_profile/layout.html.twig [hunk 8]
```diff
@@ -0,0 +1,27 @@
+{% extends 'base.html.twig' %}
+
```
Comment (AnkDev) [inline templates/user_profile/layout.html.twig]: T_T

File: templates/user_profile/layout.html.twig [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+                        <a href="{{ path('new_password') }}" class="nav-link {% if is_current_path('new_password') %} active {% endif %}">Change password</a>
+                    </li>
+                </ul>
+                <div class="tab-content py-4">
+                    <div class="tab-pane active profilepage" id="edit">
+                        {% block page_content %} {% endblock %}
+                    </div>
+                </div>
+            </div>
+
```
Comment (AnkDev) [inline templates/user_profile/layout.html.twig]: ._. Delete

File: templates/user_profile/password.html.twig [hunk 10]
```diff
@@ -0,0 +1,17 @@
+{% extends 'user_profile/layout.html.twig' %}
+
```
Comment (AnkDev) [inline templates/user_profile/password.html.twig]: :)

File: templates/user_profile/password.html.twig [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+            <label class="col-lg-3 col-form-label form-control-label"></label>
+            <div class="col-lg-9">
+                <input type="reset" class="btn btn-secondary" value="Cancel">
+                <button type="submit" class="btn btn-primary float-right"> Save Changes</button>
+            </div>
+        </div>
+    {{ form_end(form) }}
+{% endblock %}
+
+
```
Comment (AnkDev) [inline templates/user_profile/password.html.twig]: two white lines :)

File: src/Controller/UserProfileController.php [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+
+    /**
+     * @Route("/profile/newpasword", name="new_password")
+     */
+    public function changePassword(Request $request,UserPasswordEncoderInterface $passwordEncoder)
+    {
+        /** @var User $user */
+        $user = $this->getUser();
+        $changePasswordModel = new ChangePassword();
+        $form = $this->createForm(ChangePasswordType::class, $changePasswordModel);
```
Comment (turboxx) [inline src/Controller/UserProfileController.php]: why are you using `$changePassworModel` what is it for?

File: src/Entity/ChangePassword.php [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+namespace App\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+use Symfony\Component\Validator\Constraints as Assert;
+use Symfony\Component\Security\Core\Validator\Constraints as SecurityAssert;
+
+/**
+ * @ORM\Entity(repositoryClass="App\Repository\ChangePasswordRepository")
+ */
+class ChangePassword
```
Comment (turboxx) [inline src/Entity/ChangePassword.php]: Why is it an Entity?

File: src/Form/ChangePasswordType.php [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+use Symfony\Component\Form\Extension\Core\Type\RepeatedType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class ChangePasswordType extends AbstractType
+{
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+        ->add('oldPassword', PasswordType::class)
```
Comment (turboxx) [inline src/Form/ChangePasswordType.php]: what does it do?

**Pull request merged**